ocw/lib/EC2.py

```diff
@@ -213,9 +207,133 @@ def parse_image_name(self, img_name):
         return self.parse_image_name_helper(img_name, regexes)
 
     def cleanup_all(self):
+        cleanup_ec2_max_snapshot_age_days = PCWConfig.get_feature_property('cleanup', 'ec2-max-snapshot-age-days',
+                                                                           self._namespace)
+        cleanup_ec2_max_volumes_age_days = PCWConfig.get_feature_property('cleanup', 'ec2-max-volumes-age-days',
+                                                                          self._namespace)
         self.cleanup_images()
-        self.cleanup_snapshots()
-        self.cleanup_volumes()
+        if cleanup_ec2_max_snapshot_age_days >= 0:
+            self.cleanup_snapshots(cleanup_ec2_max_snapshot_age_days)
+        if cleanup_ec2_max_volumes_age_days >= 0:
+            self.cleanup_volumes(cleanup_ec2_max_volumes_age_days)
+        if PCWConfig.getBoolean('cleanup/vpc_cleanup', self._namespace):
+            self.cleanup_uploader_vpcs()
+
+    def delete_vpc(self, region, vpc, vpcId):
+        try:
+            self.log_info('{} has no associated instances. Initializing cleanup of it', vpc)
+            self.delete_internet_gw(vpc)
+            self.delete_routing_tables(vpc)
+            self.delete_vpc_endpoints(region, vpcId)
+            self.delete_security_groups(vpc)
+            self.delete_vpc_peering_connections(region, vpcId)
+            self.delete_network_acls(vpc)
+            self.delete_vpc_subnets(vpc)
+            if self.dry_run:
+                self.log_info('Deletion of VPC skipped due to dry_run mode')
+            else:
+                # finally, delete the vpc
+                self.ec2_resource(region).meta.client.delete_vpc(VpcId=vpcId)
+        except Exception as e:
+            self.log_err("{} on VPC deletion. {}", type(e).__name__, traceback.format_exc())
+            send_mail('{} on VPC deletion in [{}]'.format(type(e).__name__, self._namespace), traceback.format_exc())
```

Comment:
This looks like a rather bad style. In general one should not catch "Exception" itself except for very rare cases. And also as the "try" covers the complete body of the method I wonder why you even handle the exception on this level and not on caller level

Reply:
deleting VPC is quite complex thing which includes several step each one can fail in several different way ( network issues , vault/password  problems , various exceptions from AWS API ) idea is that any issues with deleting one VPC should not interrupt process of deleting others . 
I am aware about the fact that catching everything is bad practice , but I don't think it is applicable here . Keep in mind that even if I will remove try/catch here same function will be called again in 1 hour https://github.com/SUSE/pcw/blob/master/ocw/lib/cleanup.py#L45 so from some perspective everything is wrapped in try/catch Exception anyway :).

Reply:
well, ok, your decision :)